### DISCUSSION TITLE
Way to query at form level then assign the values which are not children of the query

### DISCUSSION THREAD

**Discussion by genogeno** [Budibase/budibase#9320] [Help]
Hi,

I have a question.  Is there a way to perform query on data provider but there are no component children under it?
But other components can get those values from the data provider?

For example:
`
* Screen
  * Form
    * Data Provider 1
    * Data Provider 2
    * Data Provider 3
    * Label A `text1 text2 text3`  (text1 data comes from data provider 1, text 2 from data provider 2, text 3 from data provider 3) 
    * Label B `text1 text2 text3`  (text1 data comes from data provider 1, text 2 from data provider 2, text 3 from data provider 3) 
`

As due to the limitations on the API, we somehow need to load this way. Is there a way to be able to load data with this configuration?  Also, this is designed this way so that we don't need to load Data Provider 1 everytime.  We need to load it once, and it can be used anywhere within the form.

**Comment by melohagan** [comment]
Hey @genogeno 

The labels need to be nested under the Data Providers to access the data.

If you don't want to have a scenario where you have three nested Data Providers, then I recommend that you build a [custom datasource](https://docs.budibase.com/docs/custom-datasource).

That way you can make you three API calls and aggregate the output. Something vaguely like:

**Custom READ method**:
```js
let response1 = await apiCall1...
let response2 = await apiCall2...
let response3 = await apiCall3...

return {
  ...response1,
  ...response2,
  ...response3
}
```

Then your screen would look like:
`
* Screen
  * Form
    * Data Provider (custom source with aggregated response)
      * Label A `text1 text2 text3` 
      * Label B `text1 text2 text3`
`
